tests/db/repositories/test_run_repository_integration.py

```
@@ -0,0 +1,410 @@
+        assert updated_run is not None
+        assert updated_run.status == RunStatus.FAILED
+        assert updated_run.completed_at is None
+    
+    def test_update_run_status_nonexistent_run(self, temp_db):
+        """Test updating status of a non-existent run raises error."""
+        repo = SQLiteRunRepository()
+        
+        with pytest.raises(RuntimeError, match="Run nonexistent_run_id not found"):
+            repo.update_run_status("nonexistent_run_id", RunStatus.COMPLETED)
+    
+    def test_list_runs_returns_all_runs_ordered(self, temp_db):
```

Ruff (0.14.6)

Unused method argument: `temp_db`

(ARG002)